src/afl-forkserver.c

```diff
@@ -95,6 +95,10 @@ void afl_fsrv_init(afl_forkserver_t *fsrv) {
   fsrv->uses_asan = false;
 
   fsrv->init_child_func = fsrv_exec_child;
+  fsrv->kill_signal = SIGKILL;
+  if (get_afl_env("AFL_KILL_SIGNAL")){
+    fsrv->kill_signal = atoi(get_afl_env("AFL_KILL_SIGNAL"));
+  }
```

Comment:
this needs error handling in the case that e.g. ` AFL_KILL_SIGNAL=foobar`  or an invalid signal number. also people are used to use the signal name instead of the signal number.

Reply:
I recommend to reset the env var with a correct value here so the same code is not needed in `at_exit()`

Reply:
Using a number was my initial idea, is there any way to get a mapping of string -> id? Personally, I'm rather used to doing kill 9

Reply:
Answering my own question, this is how kill does it:
https://github.com/coreutils/coreutils/blob/00ea4bacf6063ccc125209d5186f8f2382c6f0d4/src/operand2sig.c#L36
But I would argue users can look up the table instead

Reply:
ok, however it has still to be checked that the env var is a valid number and FATAL if not.

Reply:
I would be fine with using just a number.

Reply:
We try not to parse env variables here, but instead set the kill signal when parsing, before starting the forkserver.